dvc/render/match.py

```diff
@@ -19,12 +20,34 @@ def get_files(plots_data: Dict) -> List:
     return sorted_files
 
 
+def _get(nested_dict: Dict, path: List[str]):
+    if len(path) < 1:
+        return None
+
+    key = path[0]
+    try:
+        content = nested_dict[key]
+    except KeyError:
+        return None
+
+    if len(path) == 1:
+        return content
+    return _get(content, path[1:])
+
+
 def group_by_filename(plots_data: Dict) -> Dict:
     files = get_files(plots_data)
-    grouped = {
-        file: dpath.util.search(plots_data, ["*", "*", file]) for file in files
-    }
-    return grouped
+    revisions = list(plots_data)
+
+    grouped: Dict[str, Dict] = defaultdict(dict)
+
+    for revision in revisions:
+        for file in files:
+            path = [revision, "data", file]
+            content = _get(plots_data, path)
+            if content:
+                dpath.util.new(grouped[file], path, content)
```

Comment:
Asuming the code above is correct, why not just
```suggestion
    for revision in revisions:
        for file in files:
            data = plots_data[revision].get("data", {})
            content = data.get(file)
            if content:
                dpath.util.new(grouped[file], path, content)
```
and reduce the complexity and not even need `_get`(not used anywhere else anyway)?